isisd/isis_sr.c

```diff
@@ -627,6 +627,71 @@ static int sr_local_block_release_label(struct sr_local_block *srlb,
 	return 0;
 }
 
+static bool sr_adj_same_subnet_ipv4(struct in_addr ipv4,
+				    struct isis_circuit *circuit)
+{
+	struct listnode *node;
+	struct prefix_ipv4 *ipv4_adj, *ipv4_circuit;
+	bool is_same = false;
+
+	ipv4_adj = prefix_ipv4_new();
+	if (!ipv4_adj)
+		return false;
+
+	for (ALL_LIST_ELEMENTS_RO(circuit->ip_addrs, node, ipv4_circuit)) {
+		ipv4_adj->prefix = ipv4;
```

Comment:
Move the ipv4_adj.prefix = ipv4 outside the loop.  No need to reset it every time.  it's alwyas the same

Reply:
ok